asr-customize-speechhints.ipynb

```diff
@@ -0,0 +1,274 @@
+       "'1-800-555-4001'"
+      ]
+     },
+     "execution_count": 10,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# Applying `$FULLPHONENUM` grammar on the input text\n",
+    "apply_hint(\"one eight hundred five five five four oh oh one\",\"$FULLPHONENUM\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "id": "96dd7885-138a-4b69-b8fb-8008c3054d2b",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "'my phone number is 1-800-555-4001'"
+      ]
+     },
+     "execution_count": 11,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "apply_hint(\"my phone number is one eight hundred five five five four oh oh one\",\"$FULLPHONENUM\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "id": "7075d97c-b29f-4a03-9dd6-a79cb71486d3",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "'my phone number is 1-800-555-4001'"
+      ]
+     },
+     "execution_count": 8,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# Specifying `$FULLPHONENUM` grammar in context\n",
+    "apply_hint(\"my phone number is one eight hundred five five five four oh oh one\",\"my phone number is $FULLPHONENUM\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "id": "71aa5028-6eb3-4fc4-b7b9-9fb0a2e2c98a",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "'I think my phone number is 1-800-555-4001'"
+      ]
+     },
+     "execution_count": 13,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "apply_hint(\"I think my phone number is one eight hundred five five five four oh oh one\",\"my phone number is $FULLPHONENUM\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "id": "6289f415-ca24-44a2-b660-fe52630f93ff",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "'my phone number is one eight hundred five five five four oh oh one'"
+      ]
+     },
+     "execution_count": 9,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# Specifying `$FULLPHONENUM` grammar in Context - Context does not match\n",
+    "apply_hint(\"my phone number is one eight hundred five five five four oh oh one\",\"my phone number is not $FULLPHONENUM\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c86e9dfd-568b-410d-af3b-fcd11b5cee03",
+   "metadata": {},
+   "source": [
+    "## Sample Grammar for Handling Alphabet Sequences\n",
+    "Let's say we need to build a grammar to support the conversion of alphabet sequences to a single word ('i b m' -> 'ibm').  (For a more detailed implementation, refer to https://github.com/nvidia-riva/speech-hints/blob/main/en/oov_class_alpha_sequence.py)"
```

Comment:
```suggestion
    "Let's say we need to build a grammar to support the conversion of alphabet sequences to a single word ('i b m' -> 'ibm').  For more information, refer to the [`oov_class_alpha_sequence.py` script](https://github.com/nvidia-riva/speech-hints/blob/main/en/oov_class_alpha_sequence.py)."
```

Reply:
Since this is an example implementation. the reference is to a more detailed implementation of the grammar.